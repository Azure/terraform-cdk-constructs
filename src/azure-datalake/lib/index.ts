export * from "./filesystem";
export * from "./path";
export * from "./datalake";
