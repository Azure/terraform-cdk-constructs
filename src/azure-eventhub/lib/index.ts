export * from "./authorization";
export * from "./eventhubcluster";
export * from "./consumer";
export * from "./instance";
export * from "./kusto-connection";
export * from "./namespace";
