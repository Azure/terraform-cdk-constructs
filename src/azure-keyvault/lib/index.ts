export * from './certificate'
export * from './key'
export * from './policy'
export * from './secret'
export * from './vault'

