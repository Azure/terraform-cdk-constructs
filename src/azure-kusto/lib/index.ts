export * from "./cluster";
export * from "./compute-specification";
export * from "./database";
