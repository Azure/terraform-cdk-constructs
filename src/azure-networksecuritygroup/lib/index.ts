export * from './preconfigured-rules';
export * from './securitygroup';
