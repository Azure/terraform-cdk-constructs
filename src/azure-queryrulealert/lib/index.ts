export * from './query-rule-alert'