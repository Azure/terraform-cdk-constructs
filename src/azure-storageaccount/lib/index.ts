export * from './account';
export * from './container';
export * from './fileshare';
export * from './queue';
export * from './table';
