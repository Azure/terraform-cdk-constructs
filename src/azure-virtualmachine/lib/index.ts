export * from './image-references';
export * from './vm';
