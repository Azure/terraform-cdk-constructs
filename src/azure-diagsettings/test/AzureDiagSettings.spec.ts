import { Testing, TerraformStack} from 'cdktf';
import 'cdktf/lib/testing/adapters/jest';
import {AzurermProvider} from "@cdktf/provider-azurerm/lib/provider";
import { AzureDiagnosticSettings } from '..';
import {AppConfiguration} from "@cdktf/provider-azurerm/lib/app-configuration";
import {exampleAzureDiagnosticSettings} from "./ExampleAzureDiagSettings";

describe('Azure Diagnostics With Defaults', () => {
  let stack: TerraformStack;
  let fullSynthResult: any;

  beforeEach(() => {
    const app = Testing.app();
    stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", {features: {}});

    const appconfig = new AppConfiguration(stack, 'testAzureApplicationConfiguration', {
      name: "myappconfigstore",
      location: "eastus",
      resourceGroupName: "myResourceGroup",
      sku: "standard",
    });

    new AzureDiagnosticSettings(stack, 'testAzureDiagnosticSettingsDefaults', {
      name: "diag-settings-test",
      targetResourceId: appconfig.id,
      storageAccountId: "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/mystorageaccount",
      metricCategories: ["AllMetrics"],
      logCategories: ["AuditEvent", "Policy"],
    });

    fullSynthResult = Testing.fullSynth(stack); // Save the result for reuse
  });

  it("renders an Azure Diagnostic Settings with defaults and checks snapshot", () => {
    expect(
      Testing.synth(stack)
    ).toMatchSnapshot(); // Compare the already prepared stack
  });

  it("check if the produced terraform configuration is valid", () => {
    expect(fullSynthResult).toBeValidTerraform(); // Use the saved result
  });

  it("check if this can be planned", () => {
    expect(fullSynthResult).toPlanSuccessfully(); // Use the saved result
  });
});


describe('Azure Diagnostics With Defaults and no categories', () => {
  let stack: TerraformStack;
  let fullSynthResult: any;

  beforeEach(() => {
    const app = Testing.app();
    stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", {features: {}});

    const appconfig = new AppConfiguration(stack, 'testAzureAppConfigurationNoCat', {
      name: "myappconfigstore",
      location: "eastus",
      resourceGroupName: "myResourceGroup",
      sku: "standard",
    });

    new AzureDiagnosticSettings(stack, 'testAzureDiagnosticSettingsDefaultsNoCat', {
      name: "diag-settings-test",
      targetResourceId: appconfig.id,
      storageAccountId: "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/mystorageaccount",
      //metricCategories: ["AllMetrics"],
    });

    fullSynthResult = Testing.fullSynth(stack); // Save the result for reuse
  });

  it("renders an Azure Diagnostic Settings with defaults and checks snapshot", () => {
    expect(
      Testing.synth(stack)
    ).toMatchSnapshot(); // Compare the already prepared stack
  });

  it("check if the produced terraform configuration is valid", () => {
    expect(fullSynthResult).toBeValidTerraform(); // Use the saved result
  });

  it("check if this can be planned", () => {
    expect(fullSynthResult).toPlanSuccessfully(); // Use the saved result
  });
});

describe('Azure Diagnostic Settings Example', () => {
  
  it("renders the Azure Diagnostic Settings example and checks snapshot", () => {
    expect(
      Testing.synth(new exampleAzureDiagnosticSettings(Testing.app(), "testAzureDiagnosticSettingsExample"))
      
    ).toMatchSnapshot();
  });

  it("check if the produced terraform configuration is valid", () => {
    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(new exampleAzureDiagnosticSettings(Testing.app(), "testAzureDiagnosticSettingsExample"))).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
  
    
    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(new exampleAzureDiagnosticSettings(Testing.app(), "testAzureDiagnosticSettingsExample"))).toPlanSuccessfully();
  });
});
