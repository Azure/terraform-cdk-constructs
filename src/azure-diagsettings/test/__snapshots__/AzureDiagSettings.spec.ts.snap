// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Azure Diagnostic Settings Example renders the Azure Diagnostic Settings example and checks snapshot 1`] = `
"{
  "data": {
    "azurerm_monitor_diagnostic_categories": {
      "testdiagsettings_diagcategories_98583667": {
        "resource_id": "\${azurerm_eventhub_namespace.testEventHubNamespace.id}"
      }
    }
  },
  "output": {
    "diag_settings_name": {
      "value": "diag-settings-test"
    },
    "event_hub_namespace_id": {
      "value": "\${azurerm_eventhub_namespace.testEventHubNamespace.id}"
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_eventhub": {
      "testEventHub": {
        "message_retention": 1,
        "name": "eventhub-test",
        "namespace_name": "\${azurerm_eventhub_namespace.testEventHubNamespace.name}",
        "partition_count": 2,
        "resource_group_name": "\${azurerm_resource_group.rg.name}"
      }
    },
    "azurerm_eventhub_namespace": {
      "testEventHubNamespace": {
        "capacity": 1,
        "location": "eastus",
        "name": "eventhub-test-namespace",
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "sku": "Basic"
      }
    },
    "azurerm_eventhub_namespace_authorization_rule": {
      "testEventHubAuth": {
        "listen": true,
        "manage": true,
        "name": "eventhub-auth-test",
        "namespace_name": "\${azurerm_eventhub.testEventHub.namespace_name}",
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "send": true
      }
    },
    "azurerm_log_analytics_workspace": {
      "log_analytics": {
        "location": "eastus",
        "name": "la-test",
        "resource_group_name": "\${azurerm_resource_group.rg.name}"
      }
    },
    "azurerm_monitor_diagnostic_setting": {
      "testdiagsettings_2256DEF2": {
        "dynamic": {
          "enabled_log": {
            "content": {
              "category": "\${enabled_log.value}",
              "retention_policy": {
                "enabled": false
              }
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.testdiagsettings_diagcategories_98583667.logs)}"
          },
          "metric": {
            "content": {
              "category": "\${metric.value}",
              "retention_policy": {
                "enabled": false
              }
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.testdiagsettings_diagcategories_98583667.metrics)}"
          }
        },
        "eventhub_authorization_rule_id": "\${azurerm_eventhub_namespace_authorization_rule.testEventHubAuth.id}",
        "eventhub_name": "\${azurerm_eventhub.testEventHub.name}",
        "log_analytics_destination_type": "AzureDiagnostics",
        "log_analytics_workspace_id": "\${azurerm_log_analytics_workspace.log_analytics.id}",
        "name": "diag-settings-test",
        "target_resource_id": "\${azurerm_eventhub_namespace.testEventHubNamespace.id}"
      }
    },
    "azurerm_resource_group": {
      "rg": {
        "location": "eastus",
        "name": "rg-test"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.66.0"
      }
    }
  }
}"
`;

exports[`Azure Diagnostics With Defaults and no categories renders an Azure Diagnostic Settings with defaults and checks snapshot 1`] = `
"{
  "data": {
    "azurerm_monitor_diagnostic_categories": {
      "testAzureDiagnosticSettingsDefaultsNoCat_diagcategories_29732E12": {
        "resource_id": "\${azurerm_app_configuration.testAzureAppConfigurationNoCat.id}"
      }
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_app_configuration": {
      "testAzureAppConfigurationNoCat": {
        "location": "eastus",
        "name": "myappconfigstore",
        "resource_group_name": "myResourceGroup",
        "sku": "standard"
      }
    },
    "azurerm_monitor_diagnostic_setting": {
      "testAzureDiagnosticSettingsDefaultsNoCat_diagsettings_7B21F237": {
        "dynamic": {
          "enabled_log": {
            "content": {
              "category": "\${enabled_log.value}",
              "retention_policy": {
                "enabled": false
              }
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.testAzureDiagnosticSettingsDefaultsNoCat_diagcategories_29732E12.logs)}"
          },
          "metric": {
            "content": {
              "category": "\${metric.value}",
              "retention_policy": {
                "enabled": false
              }
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.testAzureDiagnosticSettingsDefaultsNoCat_diagcategories_29732E12.metrics)}"
          }
        },
        "log_analytics_destination_type": "AzureDiagnostics",
        "name": "diag-settings-test",
        "storage_account_id": "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/mystorageaccount",
        "target_resource_id": "\${azurerm_app_configuration.testAzureAppConfigurationNoCat.id}"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.66.0"
      }
    }
  }
}"
`;

exports[`Azure Diagnostics With Defaults renders an Azure Diagnostic Settings with defaults and checks snapshot 1`] = `
"{
  "data": {
    "azurerm_monitor_diagnostic_categories": {
      "testAzureDiagnosticSettingsDefaults_diagcategories_F2065BB8": {
        "resource_id": "\${azurerm_app_configuration.testAzureApplicationConfiguration.id}"
      }
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_app_configuration": {
      "testAzureApplicationConfiguration": {
        "location": "eastus",
        "name": "myappconfigstore",
        "resource_group_name": "myResourceGroup",
        "sku": "standard"
      }
    },
    "azurerm_monitor_diagnostic_setting": {
      "testAzureDiagnosticSettingsDefaults_diagsettings_A338353D": {
        "dynamic": {
          "enabled_log": {
            "content": {
              "category": "\${enabled_log.value}",
              "retention_policy": {
                "enabled": false
              }
            },
            "for_each": [
              "AuditEvent",
              "Policy"
            ]
          },
          "metric": {
            "content": {
              "category": "\${metric.value}",
              "retention_policy": {
                "enabled": false
              }
            },
            "for_each": [
              "AllMetrics"
            ]
          }
        },
        "log_analytics_destination_type": "AzureDiagnostics",
        "name": "diag-settings-test",
        "storage_account_id": "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/mystorageaccount",
        "target_resource_id": "\${azurerm_app_configuration.testAzureApplicationConfiguration.id}"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.66.0"
      }
    }
  }
}"
`;
